client/components/duration/duration-filter.js

```diff
@@ -0,0 +1,49 @@
+angular.module('managerApp').filter('duration', [
+  '$translate',
+  '$filter',
+  ($translate, $filter) => {
+    const unitHash = {
+      m: 'month',
+      d: 'day',
+      j: 'day',
+      y: 'year',
+      a: 'year',
+    };
+    const simpleDurationReg = /(^[0-9]+)([mdjya]?)$/;
+    const upto = /^upto/;
+    const uptoDuration = /(^upto-)([0-9]{4}-[0-9]{2}-[0-9]{2}?$)/;
+    const engage = /(^engage)([0-9]+)([mdjya]?)$/;
+
+    return (duration, dateFormat) => {
+      let d;
+      let unit;
+
+      if (simpleDurationReg.test(duration)) {
+        d = +duration.match(simpleDurationReg)[1];
+        unit = unitHash[duration.match(simpleDurationReg)[2] || 'm'];
+        return d > 1 ? $translate.instant(`common_duration_${unit}_other`, {
+          duration: d,
+        }) : $translate.instant(`common_duration_${unit}_1`);
+      }
+      if (upto.test(duration)) {
+        if (uptoDuration.test(duration)) {
+          [, , d] = duration.match(uptoDuration);
+          return $translate.instant('common_duration_upto', {
+            duration: dateFormat ? $filter('date')(d, dateFormat) : d,
```

Comment:
https://github.com/angular-translate/angular-translate/issues/1101 Ongoing issue with `$translate` and `utf8` characters , so you might want to check durations with accents display  correctly